### PR TITLE
Allow configuration to pass in a custom_search regex

### PR DESCRIPTION
**Pull request by sloped**
Allows a configuration variable of customSearch to be passed so template languages that have helpers for scripts and styles can still utilize usemin. 

```js
    useminPrepare: {
  			production: {
  				src: ['<%= views %>/**/*.blade.*'],
  				options : {
  					root: '<%= htdocs %>',
  					staging: '.tmp',
  					dest: '<%= htdocs %>',
  					customSearch: /{!!\s+HTML::(style|script)\('(\S+)'\)\s+!!}/
  				},
  			}

		},
```

### CODE REVIEW

**Comment by addyosmani** [comment]
It would be useful for new configuration settings to also include accompanying unit tests. cc @XhmikosR @sindresorhus for usefulness of the feature.

**Comment by sloped** [comment]
Unit test and documentation added.

File: README.md [hunk 1]
```diff
@@ -183,6 +183,21 @@ Default: `nil`
 
 The root directory from which your files will be resolved.
 
+### customSearch
+
+Type: 'regex'
+Default: `nil`
```
Comment (XhmikosR) [inline README.md]: `nil` is wrong.

File: README.md [hunk 2]
```diff
@@ -183,6 +183,21 @@ Default: `nil`
...
+
+A custom regular expression for creating configurations based on scripts and styles included using non standard methods. 
+
+For example:
+
+The Blade template language in Laravel 5 generated script tags with this syntax. 
+      
+          {!! HTML::script('scripts/vendor/bootstrap/bootstrap-button.js') !!}
```
Comment (XhmikosR) [inline README.md]: Too much indentation. Prefer fenced code blocks.

**Comment by XhmikosR** [comment]
1. The patches needs to be squashed; they are for the same thing
2. Windows tests fail but that is something with AppVeyor I assume. We'll know after you squash and push.

For the patch itself I can't really comment. I mean, I haven't had the need for something like that. I guess it's a valid use case, although, not sure how many will make use of it.

**Comment by sloped** [comment]
I was uncertain of how to squash commits on a current pull request. Sorry. 

Closing this and opened new #533

File: README.md (hunk 2 above)
Comment (sloped) [inline README.md]: Fixed on #533

**Comment by sindresorhus** [comment]
> I was uncertain of how to squash commits on a current pull request. Sorry.

Instead of doing it incorrectly every time, wouldn't it be better to actually take the time to learn how to do it correctly.

**Pull request closed without merging**